6.3.long_cat_is_long.py

```diff
@@ -0,0 +1,21 @@
+def count_words(text):
+    """
+    Counts the length of each word in a given text and returns a dictionary of the lengths and their frequencies.
+    :param text: The text to count the word lengths from.
+    :return: A dictionary of the lengths of each word and their frequencies.
+    """
+    # Clean text from non-alphabetic characters using a generator expression
+    words = (word.strip().strip('.,!?') for word in text.split())
```

Comment:
This is not a generic way to do it.. what if my text had ':' or '-'?
please make your code more generic